x/stake/keeper.go

```diff
@@ -469,3 +469,67 @@ func (k Keeper) setPool(ctx sdk.Context, p Pool) {
 	store.Set(PoolKey, b)
 	k.gs = Pool{} // clear the cache
 }
+
+//__________________________________________________________________________
+
+// Implements ValidatorSetKeeper
+
+var _ sdk.ValidatorSetKeeper = Keeper{}
+
+func (k Keeper) Validators(ctx sdk.Context) []*sdk.Validator {
+	vals := k.GetValidators(ctx)
+	res := make([]*sdk.Validator, len(vals))
+
+	for i, val := range vals {
+		abcival := val.abciValidator(k.cdc)
+		res[i] = &abcival
+	}
+
+	return res
+}
+
+func (k Keeper) Size(ctx sdk.Context) int {
+	return len(k.GetValidators(ctx))
+}
+
+func (k Keeper) IsValidator(ctx sdk.Context, addr sdk.Address) bool {
+	for _, v := range k.GetValidators(ctx) {
```

Comment:
Don't use loops, this functionality already exists in the keeper.

Reply:
`GetCandidate` will be also used here